### ISSUE TITLE
Meteor 1.5.2.1: uglify cannot uglify emojione

### ISSUE THREAD

**Issue by acemtp** [meteor/meteor#9138]
Hello,

Just move from 1.5.1 to 1.5.2.1 and now, deployed version failed because of new version uglify that failed to uglify emojione:

```
4049afb95685ab66e1e11da2cff24e771534ea2e.js?meteor_js_resource=true:181 Uncaught SyntaxError: Invalid regular expression: /<object[^>]*>.*?</object>|<span[^>]*>.*?</span>|<(?:object|embed|svg|img|div|span|p|a)[^>]*>|(?:🏳)️?‍?(?:🌈)|(?:👁)️?‍?(?:🗨)️?|[#-9]️?⃣|(?:(?:🏴)(?:�[�-�]){1,6})|(?:�[�-�]){2}|(?:(?:�[��]))️?(?:�[�-�])?‍?(?:[⚕⚖✈]|�[�-�]|�[������])|(?:�[��]|�[�-�])(?:�[�-�])?‍?[♀♂⚕⚖✈]?️?|(?:(?:❤|�[�-��])[‍️]{0,2}){1,3}(?:❤|�[�-��])|(?:(?:❤|�[�-��])️?){2,4}|(?:�[����-���-�]|�[��]|�[�-�]|�[�-�]|�[�-�]|⛹|👯)️?(?:�[�-�])?‍?[♀♂]?️?|(?:[☝⛹✊-✍]|�[�-�]|�[�-��-��-��-�]|�[�-�])️?(?:�[�-�])|(?:[↔-↙↩-↪]️?|[#*]|[〰〽]️?|(?:�[�-�]|🆎|�[�-�])️?|Ⓜ️?|[㊗㊙]️?|(?:�[�-�]|🈚|🈯|�[�-�]|�[�-�])️?|[‼⁉]️?|[▪-▫▶◀◻-◾]️?|[©®]️?|[™ℹ]️?|🀄️?|[⬅-⬇⬛-⬜⭐⭕]️?|[⌚-⌛⌨⏏⏩-⏳⏸-⏺]️?|🃏|[⤴⤵]️?)|[✀-➿]️?|[�-�][�-�]️?|[☀-⛿]️?|[0-9]️/: Range out of order in character class
    at RegExp (<anonymous>)
    at f.emojione (4049afb95685ab66e1e11da2cff24e771534ea2e.js?meteor_js_resource=true:181)
    at 4049afb95685ab66e1e11da2cff24e771534ea2e.js?meteor_js_resource=true:181
    at 4049afb95685ab66e1e11da2cff24e771534ea2e.js?meteor_js_resource=true:181
    at 4049afb95685ab66e1e11da2cff24e771534ea2e.js?meteor_js_resource=true:181
```

To reproduce, create a meteor project, install emojione:emojione@3.1.0 and deploy, you'll see the error in the client dev console.

Another project had this issue and they changed the uglify option to fix it, perhaps you could do that too:

https://github.com/jcbrand/converse.js/issues/920

**Comment by abernix** [comment]
Thanks for reporting this. That thread doesn't make it very clear _what_ the fix was. However, try swapping your `standard-minifier-js` package with `abernix:standard-minifier-js` and see if it resolves the problem.  The latter package includes [`uglify-es`](//npm.im/uglify-es) instead of [`uglify-js`](//npm.im/uglify-js) (its predecessor).  If it fixes it for you, the change in that update should be the default behavior in Meteor 1.6 (unless it causes other problems).

(p.s. The GitHub notification e-mail I received about this was so colorful! 🙈 🌈 )
<img width=200 src="https://user-images.githubusercontent.com/841294/30908163-76c19b24-a385-11e7-8331-342ccc301f7b.png"/>

**Comment by klaussner** [comment]
`uglify-es` has the same bug, unfortunately. :slightly_frowning_face: It looks like the minifier incorrectly replaces some escaped Unicode characters, such as `\uDC71`, with [`REPLACEMENT_CHARACTER`](https://apps.timwhitlock.info/unicode/inspect?s=%EF%BF%BD) (`\uFFFD`), which causes a character range in an [`emojione` `RegExp`](https://github.com/emojione/emojione/blob/4a9d8f3a7496bb23d352bce93f323127fdfd12de/lib/js/emojione.js#L160) to become invalid. I'm not sure if this is really new bug in `uglify-js` (although the error is thrown since `uglify-js` was updates to 3.0.28 in Meteor) because:

Input:
```js
new RegExp('\uDC6E\uDC71');
```

Output of `uglify-es` 3.1.2 and `uglify-js` 3.0.28:
```js
new RegExp("\udc6e�");
```

Output of `uglify-js` 3.0.18:
```js
new RegExp("��");
```

In other words: the version of `uglify-js` that was previously used in Meteor's minifier replaced more escaped Unicode characters with `\uFFFD` and, therefore, didn't cause the character range syntax error.

**Comment by acemtp** [comment]
I meteor removed standard-minifier-js and meteor added abernix:standard-minifier-js and it works.

**Comment by klaussner** [comment]
@acemtp I think it only works because `abernix:standard-minifier-js` uses an older version of `uglify-es` (3.0.18), which outputs the same code as `uglify-js` 3.0.18 (used in Meteor 1.5.1) for the example in my previous comment.

**Comment by acemtp** [comment]
Yes @klaussner I think it's the reason, but at least for now, it fixed my problem ;-)

**Comment by mroftalp** [comment]
I have the same error as well with my app that uses emojione. I also get errors from different web apps (non-https) like this `Uncaught SyntaxError: missing ) after argument list` in production when I try to open in mobile (Chrome device toolbar), but in desktop is ok, locally mobile and desktop is okay as well. Reverting to 1.5 fixed the problem for now.

**Comment by hwillson** [comment]
Just a quick FYI - PR https://github.com/meteor/meteor/pull/9368 fixes this issue (by bumping to the latest version of `uglify-es`).
